[Rox-9435] : Always perform a dry-run even if the policy is marked disabled

charmik-redhat opened the pull request:
## Description

Updated `dryrun` APIs to generate preview violations for disabled policies. For disabled policies, 'Preview violations' in the UI will show an info alert indicating that the policy is disabled.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [x] Evaluated and added CHANGELOG entry if required
- [ ] ~Determined and documented upgrade steps~
- [ ] ~Documented user facing changes (create PR based on [stackrox/openshift-docs](https://github.com/stackrox/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

## Testing Performed
- CI testing
- Added a new e2e test
- Manual testing on UI by editing a disabled policy

![80067DAA-68E8-4575-8F20-7E45327F3A97_1_105_c](https://user-images.githubusercontent.com/101146970/164512258-4f4e558c-d38c-4d65-be2d-b385d54459d5.jpeg)

![BECE6BE4-057F-4186-8F66-7B82C0313ED8_1_105_c](https://user-images.githubusercontent.com/101146970/164512298-1464f677-be01-4c7e-9b99-0976794a647b.jpeg)

alanonthegit commented:
hey @charmik-redhat , do we only display this message conditionally if the policy state is disabled? In any case, you could try to wrap it into an inline `info` alert component if you want to draw more attention to it. Space is already limited though, so it might be worth truncating the text description where possible. 

https://www.patternfly.org/v4/components/alert#inline-types

<img width="871" alt="image" src="https://user-images.githubusercontent.com/28272783/163496775-2116a982-55f8-448f-a8d4-12f244f4755e.png">

md2119 commented on CHANGELOG.md:
```diff
@@ -23,6 +23,7 @@ Please avoid adding duplicate information across this changelog and JIRA/doc inp
 - ROX-10038: Removed limit of 10 inclusions and 10 exclusions from policy form
 - ROX-10090: Made the username and password optional on the Artifactory integration form
 - ROX-10217: Remove format validation from the URL field of the generic webhook integration form
+- ROX-9435: Updated dryrun API to generate preview violations for disabled policies
```
Be explicit about which dry-run API, as in system policy dry-run API.

md2119 replied:
Also indicate the exact rest endpoint `/v1/policies/dryrun`

charmik-redhat replied:
Thanks, changed to system policy dry-run API.

md2119 reviewed (approved):
Minor nit, otherwlse the changes look good.

Whatever is the docs process, this fix should come out in the release notes under bug fixes. Make sure to check that when the docs are opened for review.

md2119 commented:
**(Not related to your PR)**

@vjwilson 
After seeing the info banner, for the first time the list of violations was very obvious. Otherwise, where the section info ends and where the violation list begins was unclear. Also, it is very ambiguous as to whether the headings are deployment names or container names? I can correctly pinpoint it only because I am familiar with the product, but for the first-time user, it is ambiguous.

viswajithiii commented on ui/apps/platform/src/Containers/Policies/PatternFly/Wizard/Step5/ReviewPolicyForm.tsx:
```diff
@@ -160,6 +160,16 @@ function ReviewPolicyForm({
+                                variant="info"
+                                title="Policy disabled"
+                            >
+                                <p>Violations will not generate unless policy is enabled</p>
```
```suggestion
                                <p>Violations will not be generated unless the policy is enabled</p>
```

viswajithiii replied:
It would be _really_ nice if this also came with a link that allowed people to enable the policy (or took them to the screen where they can), but that requires more intricate JS than it's probably worth...

charmik-redhat replied:
Yeah, I think adding a button there to enable the policy could be a separate UI task.

alanonthegit commented:
@vjwilson @md2119  I share the concern. I didn't comment on it at the time because I think this requires a closer look and clean up (events in general, both in the dry-run and under a violation). Zhenpeng was working on a concept here but it would be more involved than what's required of this PR.

Something short-term might be to place a divider line between the description text and the results list, and either rename the panel section to "Preview [deployment] violations", or place a header below the divider line that says "Deployment results".

At some point, we may want to consider moving to a table pattern or something more concise but we have to standardize event data to some degree before we can do so.

Also the sidepanel was meant to have its own independent scrollbar and act as a floating panel in smaller breakpoints. We didn't communicate this well enough so no worries there. Let us come up with a better approach for managing events and circle back with the team.

vjwilson commented:
> Something short-term might be to place a divider line between the description text and the results list, and either rename the panel section to "Preview [deployment] violations", or place a header below the divider line that says "Deployment results".
> 
@charmik-redhat , 
Would you like help with this, or would you prefer to take this on and get review/feedback from someone on the UI team?

charmik-redhat commented:
> > Something short-term might be to place a divider line between the description text and the results list, and either rename the panel section to "Preview [deployment] violations", or place a header below the divider line that says "Deployment results".
> 
> @charmik-redhat , Would you like help with this, or would you prefer to take this on and get review/feedback from someone on the UI team?

@vjwilson I have added a divider line and a header below the divider line stating "Deployment results". LMK if that looks good. I have added you as a reviewer.

vjwilson reviewed (approved):
/lgtm
